gips/data/sentinel1/sentinel1.py

```diff
@@ -0,0 +1,417 @@
+
+            downloader.download_all()
+            # TODO: check they are complete before moving them
+            for scene in scenes:
+                print('downloaded', scene)
+                shutil.move(os.path.join(downloaddir, scene + '.zip'),
+                            os.path.join(stagedir, scene + '.zip'))
+
+        # all the downloaded zip scenes are present
+        print('all the downloaded zip scenes are present')
+
+        scenefiles = []
+        for scene in scenes:
+
+            scenefile = os.path.join(stagedir, scene + '.tif')
+            scenefiles.append(scenefile)
+
+            if not os.path.exists(scenefile):
+
+                sourcefile = os.path.join(stagedir, scene + '.zip')
+                targetfile = os.path.join(downloaddir, scene + '.tif')
+                print('processing', sourcefile, targetfile)
+
+                cmd = "/usr/bin/gpt {} -e -SsourceProduct={} -PtargetProduct={} -Dsnap.dataio.bigtiff.compression.type=LZW".format(
+                    graph_xml_path, sourcefile, targetfile)
+                print(cmd)
+                command(cmd)
+
+                print('moving', targetfile, scenefile)
+                shutil.move(targetfile, scenefile)
+
+        # all the downloaded zips have been processed and are present
+        print('all the downloaded zips have been processed and are present')
+
+        # TODO: why is this the only way I can get the current tile ID?
+        tileid = os.path.splitext(download_fp)[0][-7:]
+        shpfile = os.path.join(stagedir, 'tile_{}.shp'.format(tileid))
+
+        outfiles = []
+        for i,scenefile in enumerate(scenefiles):
+            print('warping', scenefile)
+
+            layername = os.path.splitext(os.path.split(shpfile)[1])[0]
+
+            outfile = os.path.splitext(download_fp)[0] + '_{}.tif'.format(i)
+
+            print('gdalwarp', outfile, scenefile)
+
+            ds = gdal.Warp(outfile, scenefile, format='GTiff', cutlineDSName=shpfile, \
+                cutlineLayer=layername,options=['COMPRESS=LZW'], cropToCutline=True, \
+                dstAlpha=False, xRes=0.0001, yRes=0.0001)
+
+            outfiles.append(outfile)
+
+        if len(outfiles) > 1:
+            # merge the results and save as an asset
+            cmd = 'gdal_merge.py -n 0 -a_nodata 0 -init 0 -o {} {}'.format(download_fp, ' '.join(outfiles))
+            args = shlex.split(cmd)
+            p = subprocess.Popen(args)
+            p.communicate()
+            if p.returncode != 0:
+                raise IOError("Expected exit status 0, got {}".format(p.returncode))
+        else:
+            # no need to merge
+            shutil.move(outfiles[0], download_fp)
+
+        return True
+
+
+class sentinel1Data(Data):
+    name = 'Sentinel1'
+    version = '0.0.1'
+    Asset = sentinel1Asset
+    inline_archive = True
+
+    # TODO: is this the correct order?
+    _productgroups = {'sigma0': ['sigma0']}
```

Comment:
What's the output tile grid look like?  Is it the same as Sentinel2?

Reply:
Since S1 doesn't have a tile system I'm using our analysis tile boundaries, but at 20 meter resolution.